Document usage of this repository

pluehne opened the pull request:
This adds a new section on using this repository as a Git submodule within paper repositories.

@rkaminsk: I now recall why I reset the paths in `.latexmkrc`. The reason is that I like to build LaTeX documents in a way that is 100 % reproducible. Having include paths for bibliography, LaTeX styles/classes, etc. configured per system is in opposition to this goal. For this reason, I decided to reset these include paths in `.latexmkrc`. This avoids situations where 1) someone is missing a style class or similar that coworkers just assume to be present on your machine or 2) someone else has an outdated version of a dependency and results look different (I’m talking about the LLNCS vs. `svproc` mess in particular).

## To Do

- [x] Add a section on setting environment variables globally instead of using `.latexmkrc` (@rkaminsk: Can you send me your configuration?)
- [x] Work around `ensure_path` on systems where the directive isn’t present

Fixes #1.

pluehne commented:
I’m starting to think that the readme is too long. I could extract the usage and contributing parts into separate documents and link to them from the actual readme. Thoughts?

rkaminsk commented:
The problem I had with your config was that I got the error message "function `ensure_path` not found". So I simply did it like this:
```perl
# Ignore any locally installed files
$ENV{"TEXMFHOME"} = "?";

# Reset all search paths
$ENV{"BIBINPUTS"} = "./include//:";
$ENV{"BSTINPUTS"} = "./include//:";
$ENV{"TEXINPUTS"} = "./include//:";
```

pluehne commented:
@rkaminsk: Your solution should also reset all paths, is that correct? In that case, it would be much cleaner than mine and achieve the same thing :slightly_smiling_face:.

rkaminsk commented:
> Your solution should also reset all paths.

I think it does the same as yours. I even think it is portable across platforms because even windows should be fine with the slashes.

pluehne commented:
@rkaminsk: I addressed your comment. You had an alternative way for solving the include path problem in mind, am I right? In this case, would you just recommend extending the `.bashrc` similar to what the `.latexmkrc` does?

rkaminsk commented:
> Add a section on setting environment globally instead of using `.latexmkrc`.

I would just tell people to set the environment variables. Give an example for bash. And note that Latex IDEs provide settings to customize these paths.

```bash
export BIBINPUTS=".//include/:"
export BSTINPUTS=".//include/:"
export TEXINPUTS=".//include/:"
```

rkaminsk commented:
I think that the length of the README is still fine.

pluehne commented:
@rkaminsk: I added a short piece of text on what to do if you’re not using `latexmk`.

@tortinator, @rkaminsk: That’s everything I had on my to-do list. Please review again and approve or comment as necessary :slightly_smiling_face:.

pluehne commented:
I renamed the `includes` directory to `include` after @MaxOstrowski pointed out that both were used inconsistently. I agree that `include` sounds more conventional than the plural form.

pluehne commented:
Also, @MaxOstrowski pointed out that 1) I had accidentally dropped the code example with `\bibliography{krr,procs}` and 2) many links to `.bib` files pointed to nowhere with the latest changes in the file structure. I just addressed both issues.

tortinator commented:
Hallo @pluehne, I close this for the time being, cheers -t
